src/lazy/text/buffer.rs

```diff
@@ -2878,8 +2914,17 @@ mod tests {
             "5.0d+1", "-5.0d-1",
         ],
         expect_mismatch: [
-            "123._456", "5", "5d", "05d", "-5d", "5.d", "-5.d", "5.D", "-5.D", "5.1d", "-5.1d",
-            "5.1D", "-5.1D", "-5.0+0",
+            "123._456", "5", "05d", "-5.0+0",
+        ],
+        expect_incomplete: [
+            "5d",
+            "-5d",
+            "5.d",
+            "-5.d",
+            "5.D",
+            "-5.D",
+            "5.1d", "-5.1d",
+            "5.1D", "-5.1D",
```

Comment:
🗺️ All of these _could_ be valid decimals if the buffer ends up getting at least one more digit.